### PR TITLE
redux&saga

### PR DESCRIPTION
**Pull request by OlegSliunchenko**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,14 +3,20 @@
...
     "@testing-library/user-event": "^13.5.0",
     "axios": "^0.26.1",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
+    "react-redux": "^8.0.1",
     "react-scripts": "5.0.0",
     "react-slick": "^0.28.1",
+    "redux": "^4.2.0",
+    "redux-actions": "^2.6.5",
+    "redux-devtools-extension": "^2.13.9",
```
Comment (Akyna) [inline package.json:18]: should be in dev dependencies

File: src/App.js [hunk 2]
```diff
@@ -1,140 +1,28 @@
...
-    setPasswordError(data.isPasswordError);
-    setAccountError(data.isAccountError);
-  };
-
-  // return <LoginPage userInfo={user} />;
-  // return <MovieInformationPage userInfo={user} movie={movieList[0]} />;
-  return <MovieListPage userInfo={user} />;
+  if (isUserAuthorized) {
+    return <MovieListPage />;
+    // return <MovieInformationPage />;
```
Comment (Akyna) [inline src/App.js:23]: Pls don't use comment(s) like this

Remove

File: src/sagas/movieSaga.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+function* movieWorker() {
+  try {
+    const {
+      data: { results },
+    } = yield call(
+      axios.get,
+      'https://my.api.mockaroo.com/movies.json?key=c1952b60'
+    );
+    yield delay(3000);
```
Comment (Akyna) [inline src/sagas/movieSaga.js:20]: what is the reason for this?

File: package.json (hunk 1 above)
Comment (OlegSliunchenko) [reply package.json:18]: reinstalled

File: src/App.js (hunk 2 above)
Comment (OlegSliunchenko) [reply src/App.js:23]: I think, on this stage of project, we need it. We didn't have any request for routing yet, but if master Yoda tells....Done

File: src/components/MovieList/index.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  errorMessageSelector,
+  isLoading,
+  movieListSelector,
+} from '../../redux/modules/movie/selectors';
+
+import { InfoMessage, ListItem, Spinner, SpinnerErrorWraper } from '../index';
```
[the remark this answers is not in the document]
Comment (OlegSliunchenko) [reply src/components/MovieList/index.js:9]: go one filder up(in to components) and take index.js

File: src/pages/MovieListPage.js [hunk 5]
```diff
@@ -130,18 +36,7 @@ export default function MovieListPage({ userInfo }) {
               <ListItemHeader />
               <ListHeaderDevider number={7} />
```
[the remark this answers is not in the document]
Comment (OlegSliunchenko) [reply src/pages/MovieListPage.js:37]: colSpanNumber

File: src/redux/modules/auth/actions.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import { createAction } from '@reduxjs/toolkit';
+
+export const LoginAction = createAction('auth/getLoginUser');
+export const isUserLogged = createAction('auth/isUserLogged');
+export const UserLoginError = createAction('auth/UserLoginError');
+export const resetButton = createAction('auth/ResetLoginInput');
```
[the remark this answers is not in the document]
Comment (OlegSliunchenko) [reply src/redux/modules/auth/actions.js:6]: what is wrong with 'is' in here?

File: src/components/MainContainer/style.css [hunk 7]
```diff
@@ -143,3 +142,11 @@ input[type="password"]::placeholder {
     align-items: center;
 
 }
+
+.spinner-center {
+    width: 100%;
```
Comment (Akyna) [inline src/components/MainContainer/style.css:147]: the percentage will be taken from parents. are you sure that the parent also has 100% of width/ height?
Maybe will be better to use vh/wh?

File: src/redux/modules/auth/actions.js [hunk 8]
```diff
@@ -0,0 +1,6 @@
+import { createAction } from '@reduxjs/toolkit';
+
+export const loginAction = createAction('auth/getLoginUser');
+export const isUserLogged = createAction('auth/isUserLogged');
```
Comment (Akyna) [inline src/redux/modules/auth/actions.js:4]: is - its value

use 

get / set / update / simple action

File: src/redux/modules/movie/actions.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+import { createAction } from '@reduxjs/toolkit';
+
+export const MovieListLoading = createAction('movie/movieListStarted');
+export const setMovieList = createAction('movie/movieListDone');
+export const MovieListError = createAction('movie/movieListError');
+export const MovieDetailsLoading = createAction('movie/currentMovie');
+export const setMovieDetails = createAction('movie/movieDetails');
+export const MovieDetailsError = createAction('movie/movieDetailsError');
```
Comment (Akyna) [inline src/redux/modules/movie/actions.js:8]: forgot maybe

loverCamelCase

CamelCase

File: src/sagas/authSaga.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+} from '../redux/modules/auth/actions';
+
+function* loginWorker() {
+  try {
+    const user = {
+      userName: 'SkySlider',
+      userId: 333,
+      userEmail: 'test@test.com',
+    };
+    yield delay(3000);
```
Comment (Akyna) [inline src/sagas/authSaga.js:15]: ?????

**Review by Akyna — APPROVED** [review]

**Pull request merged**